vumi/components/message_store.py

```diff
@@ -431,12 +440,18 @@ def add_event(self, event):
             event_record = self.events(event_id, event=event, message=msg_id)
         else:
             event_record.event = event
-        yield event_record.save()
 
-        msg_record = yield self.outbound_messages.load(msg_id)
-        if msg_record is not None:
-            for batch_id in msg_record.batches.keys():
-                yield self.cache.add_event(batch_id, event)
+        # If we aren't given batch_ids, get them from the outbound message.
+        if batch_ids is None:
+            msg_record = yield self.outbound_messages.load(msg_id)
+            if msg_record is not None:
+                batch_ids = msg_record.batches.keys()
+
+        for batch_id in batch_ids:
```

Comment:
If `msg_record` is `None` above, wouldn't `batch_ids` still be `None` at this point?